### PR TITLE
12.0 legalsylvain account customer wallet

### PR DESCRIPTION
**Pull request by legalsylvain**
Hi @carmenbianca.

i propose some changes in the ``account_customer_wallet`` to 
- cover the use case when a customer has a positive wallet and want to mark as paid a pending invoice.
- be able to prevent invoice payments with wallet if wallet balance is not enough.

image

image

**Note**
1) I have a local problem with pre-commit, so I disabled a test in the ``.pre-commit-config.yaml`` file. Iet me know when you want to merge this PR (if you want !) and I'll remove this commit. 
2) i changed the partner adress to a partner in the test. In fact, there is a problem in the computation of the wallet balance I think. As other "accounting" data, the wallet balance, should be at the parent partner level. 
given the following configuration : 
- parent partner A : 
- - partner Adress A1
- - partner Adress A2
if we credit debit A1 or A2, it should in my opinion recompute the wallet of the partner A.

image

### CODE REVIEW

**Review by carmenbianca — APPROVED** [review]
LGTM! Thank you @legalsylvain 

> In fact, there is a problem in the computation of the wallet balance I think. As other "accounting" data, the wallet balance, should be at the parent partner level.

This is a known issue. I'm not sure how much time we want to spend fixing this.

**Comment by legalsylvain** [comment]
> This is a known issue. I'm not sure how much time we want to spend fixing this. 

do you want I make a try in the same branch ?

i can also try to do the works regarding that comments (https://github.com/coopiteasy/addons/pull/222#pullrequestreview-949838483) if you think the comments are relevant ? 

if yes, let me know, and please say me a time slot when you don't work on that project, to avoid git conflict.

**Comment by legalsylvain** [comment]
Note : as it is the first time I contribute to your repo, the CI will wait your approval.
image

The approval will be required each time i commit again.

once the PR merged, you'll not have to do it again.

**Comment by carmenbianca** [comment]
Thanks so much for the valuable communication @legalsylvain 

> do you want I make a try in the same branch ?

I think it's best to do this in a separate branch. But it doesn't matter that much and I leave it to you.

> i can also try to do the works regarding that comments (https://github.com/coopiteasy/addons/pull/222#pullrequestreview-949838483) if you think the comments are relevant ?

I'll do these :) Thank you for offering.

**Comment by legalsylvain** [comment]
Hi. Thanks for your review. I removed the "pre-commit" commit. 
this PR could be so merged if you're OK

regards.

**Comment by legalsylvain** [comment]
rebased, could be merged I think.

**Pull request merged**
